packages/grpc-js-core/src/call-stream.ts

```diff
@@ -219,14 +262,13 @@ export class Http2CallStream extends Duplex implements CallStream {
         }
       });
       stream.on('trailers', this.handleTrailers.bind(this));
-      stream.on('data', (data) => {
+      stream.on('data', (data: Buffer) => {
         let readHead = 0;
-        let canPush = true;
         let toRead: number;
         while (readHead < data.length) {
           switch (this.readState) {
             case ReadState.NO_DATA:
-              this.readCompressFlag = (data.readUInt8(readHead) !== 0);
+              this.readCompressFlag = data.slice(readHead, readHead+1);
```

Comment:
Very minor nit: not sure this is passing linter properly due to the `+1` with no spaces around it.